src/targets/maven.ts

```diff
@@ -0,0 +1,172 @@
+};
+
+/**
+ * Target responsible for uploading files to Maven Central.
+ */
+export class MavenTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'maven';
+  /** Target options */
+  public readonly mavenConfig: MavenTargetConfig;
+  /** GitHub client. */
+  public readonly github: Github;
+  /** GitHub remote. */
+  public readonly githubRemote: GithubRemote;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.mavenConfig = this.getMavenConfig();
+    this.github = getGithubClient();
+    this.githubRemote = new GithubRemote(GIT_REPO_OWNER, GIT_REPO_NAME);
+  }
+
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      // Required env variables
+      ossrhUsername: this.getEnvVarValue('OSSRH_USERNAME'),
+      ossrhPassword: this.getEnvVarValue('OSSRH_PASSWORD'),
+      // TODO: MAVEN_CENTRAL_* shouldnt be required if the user already has `gradle.properties`
+      mavenUsername: this.getEnvVarValue('MAVEN_CENTRAL_USERNAME'),
+      mavenPassword: this.getEnvVarValue('MAVEN_CENTRAL_PASSWORD'),
+      // Optional env variables
+      distributionsPath: this.getEnvVarOrDefault('DISTRIBUTIONS_PATH'),
+      settingsPath: this.getEnvVarOrDefault('SETTINGS_PATH'),
+      mavenRepoUrl: this.getEnvVarOrDefault('MAVEN_REPO_URL'),
+      mavenRepoId: this.getEnvVarOrDefault('MAVEN_REPO_ID'),
+      mavenCliPath: this.getEnvVarOrDefault('MAVEN_CLI_PATH'),
+      gradleCliPath: this.getEnvVarOrDefault('GRADLE_CLI_PATH'),
+    };
+  }
+
+  private getEnvVarValue(envVar: string): string {
+    if (process.env[envVar]) {
+      return process.env[envVar] as string; // `as string` to make TS happy
+    }
+    throw new ConfigurationError(
+      `Cannot publish to Maven Central: missing credentials.
+      Please, use the ${envVar} environment variable.`
+    );
+  }
+
+  private getEnvVarOrDefault(envVar: string): string {
+    if (process.env[envVar]) {
+      return process.env[envVar] as string; // `as string` to make TS happy
+    }
+    return DEFAULT_ENV_VARIABLES[envVar];
+  }
+
+  public async publish(version: string, _revison: string): Promise<void> {
+    await withTempDir(
+      async dir => {
+        const git = simpleGit(dir);
+        const username = await getAuthUsername(this.github);
+        this.githubRemote.setAuth(username, getGithubApiToken());
+        await git.clone(this.githubRemote.getRemoteStringWithAuth(), dir);
+        await git.checkout(`release/${version}`); // TODO: release name should be customized
+
+        await this.createUserGradlePropsFile();
+        const mavenReleaser = new MavenReleaser(
+          this.mavenConfig.distributionsPath,
+          this.mavenConfig.settingsPath,
+          this.mavenConfig.mavenRepoUrl,
+          this.mavenConfig.mavenRepoId,
+          this.mavenConfig.mavenCliPath,
+          this.mavenConfig.gradleCliPath
+        );
```

Comment:
This indicates that `MavenReleaser` should not be a separate class as you are passing so many arguments related to the current instance to this class. It seems like this should just be part of the current target class?